File: healthcare/src/main/java/com/example/zero/healthcare/dto/journal/UpdateJournalRequest.java
package com.example.zero.healthcare.dto.journal;

import jakarta.validation.Valid;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.validator.constraints.URL;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
public class UpdateJournalRequest {

    @Valid
    private PreConditionDto preCondition;

    @Valid
    private PostConditionDto postCondition;

    @Valid
    private List<PainRecordDto> prePainRecords;

    @Valid
    private List<PainRecordDto> postPainRecords;

    @Size(max = 5000)
    private String content;

    @Valid
    @Size(max = 5)
    private List<@URL @Size(max = 1024) String> imageUrls;
}

File: HealthCareApp/app/src/main/java/com/example/healthcareapp/data/UpdateJournalRequest.kt
package com.example.healthcareapp.data

data class UpdateJournalRequest(
    val preCondition: PreConditionDto? = null,
    val postCondition: PostConditionDto? = null,
    val prePainRecords: List<PainRecordDto>? = null,
    val postPainRecords: List<PainRecordDto>? = null,
    val content: String? = null,
    val imageUrls: List<String>? = null
)
